pytorch_lightning/trainer/properties.py

```diff
@@ -484,6 +484,13 @@ def sanity_checking(self, val: bool) -> None:
         elif self.sanity_checking:
             self.state.stage = None
 
+    def loops_state_dict(self) -> Dict[str, Any]:
+        return {
+            "fit_loop": self.fit_loop.state_dict(),
+            "validate_loop": self.validate_loop.state_dict(),
+            "test_loop": self.test_loop.state_dict(),
+        }
+
```

Comment:
why does this need to be exposed as  a trainer property?

Reply:
Hey @ananthsub,

1. This isn't a strong requirements. In CheckpointConnector, we would have

```
def get_trainer_states(self):

    return {
        "loops": self.trainer.loops_state_dict(),
        ....
    }
```

2. I would say it is the target to save the `progress_tracking Dataclasses` + Results objects to enable full fail-tolerant training.

Reply:
In that case I'd prefer not exposing this now, as it's only relevant for checkpointing. We could introduce this later if we absolutely need to

Reply:
I would also prefer to  just let checkpoint connector directly access the loops to get the state dict, for now. 
We do the same for callbacks, precision, etc.

Reply:
```suggestion
```